### ISSUE TITLE
How does this compare to other vector graphics rendering libraries performance wise? 

### ISSUE THREAD

**Issue by colbyn** [audulus/vger-rs#4]
# TLDR

If I may ask, I'm currently using Google's Skia library for vector graphics rendering VIA the [skia-safe](https://crates.io/crates/skia-safe) crate, but I'd love to use a more 'rusty' solution. Do you have any idea how this would compare to Google's Skia library performance wise? As well as other popular rust crates? 

# Overview 

I happen to be in the market for a fast 2D vector graphics rendering library (for my [subscript project](https://github.com/subscript-publishing/subscript)). Specially for the canvas renderer shown in the image:  

![Canvas Preview](https://user-images.githubusercontent.com/2118601/199102721-c6d270a6-cdd9-446b-bd33-1462c1e0205a.PNG)

All I care about are paths with lines, bezier curves, and fill/stroke support. And I'm kinda interested in [rui](https://github.com/audulus/rui) so I though I'd maybe consider this library...

**Comment by wtholliday** [comment]
vger does almost everything on the GPU, so it should compare favorably to CPU-based renderers (I believe skia is cpu). My motivation was to beat nanovg (which is partially GPU) for my own use case.

You could try vger for your application, but I make no performance guarantees. I haven't tested performance of strokes, which is what you'd need. Furthermore, with vger you'd have to render a lot of line segments (vger doesn't have compound strokes), and the AA where the segments overlap wouldn't be correct compared to a single stroke.

best

**Comment by colbyn** [comment]
@wtholliday Oh thanks for the reply! I should have mentioned that I'm using Skia VIA their Metal backend which supports GPU rendering, which is why I was curious if you've considered it as an alternative to nanovg. 

PS
While it's kinda messy, I wrote up a small tutorial [over here](https://github.com/rust-skia/rust-skia/discussions/552) on how to embed the [skia-safe](https://crates.io/crates/skia-safe) crate in an IOS app using Skia's metal backend, if you ever want to try it...

**Comment by wtholliday** [comment]
Ah ok... I don't know to what extent Skia uses the GPU. I get the feeling that if they were particularly advanced I would have heard about it. I did a quick look through their code for metal stuff and nothing jumped out.
